src/rust/ru.readmanga/src/parser.rs

```diff
@@ -315,12 +315,17 @@ pub fn parse_chapters(html: &WNode, manga_id: &str) -> Result<Vec<Chapter>> {
 pub fn get_page_list(html: &WNode) -> Result<Vec<Page>> {
 	let parsing_error = helpers::create_parsing_error();
 
-	let script_text = html
+	// aidoku::prelude::println!("get_pages_html: {}", html.text());
+	let mut script_text = html
 		.select(r"div.reader-controller > script[type=text/javascript]")
 		.pop()
 		.map(|script_node| script_node.data())
 		.ok_or(parsing_error)?;
 
+	// removing first part of scirpt containing unnecessary links for reader
+	script_text.replace_range(0..script_text.find("rm_h.readerDoInit(").unwrap_or_default(), "");
```

Comment:
this doesn't need to be a mutable variable and an extra statement, you can just chain it on after the `ok_or`

Reply:
Hey, refactored, thank you.
I'm not an experienced rust programmer, so thank you for refactoring ideas.